include/oneapi/dpl/pstl/hetero/dpcpp/utils_ranges_sycl.h

```diff
@@ -277,6 +221,16 @@ struct is_hetero_legacy_trait<Iter, ::std::enable_if_t<Iter::is_hetero::value>>
 template <typename Iter>
 inline constexpr bool is_sycl_iterator_v = is_sycl_iterator<Iter>::value || is_hetero_legacy_trait<Iter>::value;
```

Comment:
Does it make sense to reimplement this like
```C++
template <typename Iter>
using is_sycl_iterator_t = std::conjunction<is_sycl_iterator<Iter>, is_hetero_legacy_trait<Iter>>;

template <typename Iter>
inline constexpr bool is_sycl_iterator_v = is_sycl_iterator_t<Iter>::value;
```

Reply:
I don't know if I agree with your exact suggestion, but I definitely don't like that `is_sycl_iterator` and  `is_sycl_iterator_v` don't mean **exactly** the same thing.
My suggestion would probably be to just rename `is_sycl_iterator_v` to be `is_sycl_or_hetero_iterator_v`. If you agree I can include it in this PR.  If we get more involved than that, it is probably outside the scope of this PR.

Reply:
Dan, about any renames - up to you.
My idea was simple define some `constexpr bool` const state as `value` from other type, nothing more.

Reply:
I'll make a type and value version of `is_sycl_or_hetero_iterator[_v]`.